### ISSUE TITLE
Bug: Does not allow node.js public class fields if ecmaVersion is less than 13

### ISSUE THREAD

**Issue by TotallyInformation** [eslint/eslint#15961]
### Environment

Node version: 16.14.0
npm version: 8.3.1
Local ESLint version: 8.16.0
Global ESLint version: 8.16.0
Operating System: Windows 10

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = {
    env: {
        browser: false,
        commonjs: true,
        jquery: false,
        node: true,
        es2019: true,  // tried with various combinations of env and parserOptions
        'shared-node-browser': false
    },
    parserOptions: {
        ecmaVersion: 12,
        sourceType: 'script'
    },
    root: true,
}
```
</details>

```js
class UibPackages {
    // TODO: Replace _XXX with #XXX once node.js v14 is the minimum supported version
    /** Flag to indicate whether setup() has been run
     * @type {boolean}
     * @protected 
     */
    _isConfigured

    constructor() {  // this gets marked by the linter as being in error
        // ....
    }
}
```

### What did you expect to happen?

According to caniusethis, public class fields were supported from node.js v12.0.0. My class requires a minimum of node.js v12.20 LTS which supports ES2019 which certainly includes them.

### What actually happened?

They work perfectly in actual operation, it is only the linter that is marking them in error.

The linter gives an error: `Parsing error: Unexpected token constructor` and there is no apparent way to override this error.

UPDATE: If >1 public class variable is declared, just the 2nd definition is marked as being in error, the constructor is then not marked in error.

UPDATE2: If the first public class variable definition is given an initial value, the `=` gets the error `Parsing error: Unexpected token =`, no other errors are given.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
it's working as expected - class fields was introduced in es2022/13.

**Comment by TotallyInformation** [comment]
> it's working as expected - class fields was introduced in es2022/13.

No it isn't and that isn't correct. Public class variables were, as I said, introduced in node.js v12

https://onecompiler.com/nodejs/3y5zhpz3d
